### PR TITLE
Support save and get GC safe point to PD

### PR DESCRIPTION
**Pull request by MyonKeminta**

### CODE REVIEW

File: server/core/kv.go [hunk 1]
```diff
@@ -234,6 +235,36 @@ func (kv *KV) LoadRegions(regions *RegionsInfo) error {
 	}
 }
 
+// SaveGCSafePoint saves new GC safe point to KV
+func (kv *KV) SaveGCSafePoint(safePoint uint64) error {
+	key := path.Join(gcPath, "safe_point")
+	value, err := json.Marshal(safePoint)
```
Comment (disksing) [inline server/core/kv.go]: How about using `strconv.FormatUint` instead?

Comment (MyonKeminta) [reply server/core/kv.go]: great!

File: server/core/kv.go [hunk 2]
```diff
@@ -234,6 +235,36 @@ func (kv *KV) LoadRegions(regions *RegionsInfo) error {
 	}
 }
 
+// SaveGCSafePoint saves new GC safe point to KV
```
Comment (nolouch) [inline server/core/kv.go]: add a comma at the end of the sentence.

Comment (MyonKeminta) [reply server/core/kv.go]: maybe period but not comma(

Comment (nolouch) [reply server/core/kv.go]: you are right.

File: server/core/kv_test.go [hunk 3]
```diff
@@ -157,6 +158,22 @@ func (s *testKVSuite) TestLoadRegionsExceedRangeLimit(c *C) {
...
+	r, e := kv.LoadGCSafePoint()
+	c.Assert(r, Equals, uint64(0))
+	c.Assert(e, IsNil)
+	for _, safePoint := range testData {
+		err := kv.SaveGCSafePoint(safePoint)
+		c.Assert(err, IsNil)
+		safePoint1, err := kv.LoadGCSafePoint()
+		c.Assert(err, IsNil)
+		c.Assert(safePoint, Equals, safePoint1)
+	}
```
Comment (disksing) [inline server/core/kv_test.go]: Any need to construct a test case that safePoint is less than previous safePoint?

Comment (disksing) [reply server/core/kv_test.go]: oh, I found it in client test.

File: pd-client/client_test.go [hunk 4]
```diff
@@ -26,6 +26,7 @@ import (
 	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/pingcap/pd/server"
 	"google.golang.org/grpc"
+	"math"
```
Comment (disksing) [inline pd-client/client_test.go]: Please reorder imports. Builtin packages should go before 3rd-party packages.

**Comment by disksing** [comment]
LGTM

File: server/core/kv.go [hunk 5]
```diff
@@ -234,6 +235,33 @@ func (kv *KV) LoadRegions(regions *RegionsInfo) error {
...
+	key := path.Join(gcPath, "safe_point")
+	value := strconv.FormatUint(safePoint, 16)
+	if err := kv.Save(key, value); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// LoadGCSafePoint loads current GC safe point from KV.
+func (kv *KV) LoadGCSafePoint() (uint64, error) {
```
Comment (siddontang) [inline server/core/kv.go]: @disksing 
do we need to record the safe point in the gauge like TSO/cluster ID does?

Comment (disksing) [reply server/core/kv.go]: TiDB already has metrics for it. Just reminded that maybe we need to set up a safe value for it when we use `pd-recover` to restore PD cluster.

**Comment by MyonKeminta** [comment]
@disksing @nolouch PTAL again

**Pull request merged**